service/src/main/kotlin/fi/espoo/evaka/reports/FamilyDaycareMealReport.kt

```diff
@@ -0,0 +1,253 @@
+                        lunchCount = row.lunchCount,
+                        snackCount = row.snackCount,
+                        childResults = childrenByDaycare[row.daycareId.toString()].orEmpty()
+                    )
+                )
+            } else if (row.areaId != null && row.areaName != null) {
+                collectedAreaResults.add(
+                    FamilyDaycareMealAreaResult(
+                        areaId = row.areaId,
+                        areaName = row.areaName,
+                        breakfastCount = row.breakfastCount,
+                        lunchCount = row.lunchCount,
+                        snackCount = row.snackCount,
+                        daycareResults = daycaresByArea[row.areaId.toString()].orEmpty()
+                    )
+                )
+            } else {
+                totalResult =
+                    FamilyDaycareMealReportResult(
+                        breakfastCount = row.breakfastCount,
+                        lunchCount = row.lunchCount,
+                        snackCount = row.snackCount,
+                        areaResults = collectedAreaResults
+                    )
+            }
+        }
+
+        return totalResult
+    }
+
+    data class FamilyDaycareMealReportRow(
+        val daycareName: String?,
+        val firstName: String?,
+        val lastName: String?,
+        val areaName: String?,
+        val childId: PersonId?,
+        val daycareId: DaycareId?,
+        val areaId: AreaId?,
+        val breakfastCount: Int,
+        val lunchCount: Int,
+        val snackCount: Int
+    )
+
+    data class FamilyDaycareMealReportResult(
+        val breakfastCount: Int,
+        val lunchCount: Int,
+        val snackCount: Int,
+        val areaResults: List<FamilyDaycareMealAreaResult>
+    )
+
+    data class FamilyDaycareMealAreaResult(
+        val areaId: AreaId,
+        val areaName: String,
+        val breakfastCount: Int,
+        val lunchCount: Int,
+        val snackCount: Int,
+        val daycareResults: List<FamilyDaycareMealDaycareResult>
+    )
+
+    data class FamilyDaycareMealDaycareResult(
+        val daycareId: DaycareId,
+        val daycareName: String,
+        val breakfastCount: Int,
+        val lunchCount: Int,
+        val snackCount: Int,
+        val childResults: List<FamilyDaycareMealChildResult>
+    )
+
+    data class FamilyDaycareMealChildResult(
+        val childId: PersonId,
+        val firstName: String,
+        val lastName: String,
+        val breakfastCount: Int,
+        val lunchCount: Int,
+        val snackCount: Int,
+    )
+
+    data class MealReportConfig(
+        val breakfastTime: MealTimeRange,
+        val lunchTime: MealTimeRange,
+        val snackTime: MealTimeRange
+    )
+    data class MealTimeRange(
+        @ForceCodeGenType(String::class) val startTime: LocalTime,
+        @ForceCodeGenType(String::class) val endTime: LocalTime
+    )
```

Comment:
There's also a reusable `TimeRange` type in `shared/domain/Time.kt`. Furthermore, the `@ForceCodeGenType` annotations are not necessary because this type doesn't end up in generated files, and `LocalTime` works with codegen anyway.